trieste/acquisition/optimizer.py

```diff
@@ -63,7 +64,9 @@ def automatic_optimizer_selector(
         return optimize_discrete(space, target_func)
 
     elif isinstance(space, (Box, TaggedProductSearchSpace)):
-        num_samples = tf.minimum(5000, 1000 * tf.shape(space.lower)[-1])
+        num_samples = tf.minimum(
+            DEFAULTS.NUM_SAMPLES_MIN, DEFAULTS.NUM_SAMPLES_DIM * tf.shape(space.lower)[-1]
+        )
```

Comment:
It is slightly problematic that a user cant just look at the optimizer and see what the default behaviour is (i.e. they would have to hop into the utils script)?

Reply:
I kinda agree but then why do we have that `DEFAULTS` object, we either use it for our defaults or not and put the constants in a file where they are used, from developer perspective same file option would be better, from user perspective and when you want to modify multiple constants it might be better to have a single object where you change default behaviour - but then in the `DEFAULTS` object we have very little at the moment

Reply:
perhaps we use it in files at the moment, I'll check all the files and if we see that we have more constants spread around I'll collect them in DEFAULTS in a separate PR

Reply:
I think the JITTER default makes sense as it isnt something that a user would really ever change and is used in loads of location across the code. This is a very specific parameter (only used here) that is likely to be changed often by users?